lib/cdo/rack/global_edition.rb

```diff
@@ -108,29 +100,54 @@ def call
 
         # Sets the request cookies to apply changes immediately without needing to reload the page.
         request.cookies[REGION_KEY] = region
-        request.cookies[LANGUAGE_COOKIE_KEY] = Cdo::GlobalEdition.region_locale(region) if region
+
+        if region
+          region_locales = Cdo::GlobalEdition.region_locales(region)
+          request.cookies[LOCALE_KEY] = region_locales.first unless region_locales.include?(request.cookies[LOCALE_KEY])
+        end
 
         # Updates the global `ge_region` cookie to lock the platform to the regional version.
-        set_global_cookie(REGION_KEY, region)
+        set_global_cookie(REGION_KEY, region, high_priority: true)
         # Updates the global `language` cookie to enforce the switch to the regional language.
-        set_global_cookie(LANGUAGE_COOKIE_KEY, request.cookies[LANGUAGE_COOKIE_KEY])
+        set_locale_cookie(request.cookies[LOCALE_KEY])
       end
 
-      private def app_route?(path)
-        Rails.application.routes.recognize_path(path).present?
+      private def dashboard_route?(path = request.path)
+        return false unless request.hostname == CDO.dashboard_hostname
+        Dashboard::Application.routes.recognize_path(path, method: request.request_method).present?
       rescue ActionController::RoutingError
         false
       end
 
+      private def pegasus_helpers
```

Comment:
where is this used?

Reply:
I missed this part after many merge conflict fixes